### ISSUE TITLE
Add instant ajax search

### ISSUE THREAD

**Issue by Dubrzr** [Dubrzr/golb#5]
Search will always return articles as results for now.
It must handle:
-> search for username or first name/last name will return articles edited by this user
-> search for words will first return articles which have tags corresponding to those words
-> search for words will return articles ordered by most relevant results
-> search for date will return articles with the exact or near date

**Comment by choucavalier** [comment]
I think the search-by-word feature should also look at the content of the articles.

I think django-haystack (http://haystacksearch.org/) is what we are looking for.

**Comment by Dubrzr** [comment]
Yep this is what I meant when saying "search for words will return articles ordered by most relevant results", it will look at the content of articles.

If Haystask search can handle all features yep, we can use it!

**Comment by choucavalier** [comment]
We could also use elastic search https://github.com/liberation/django-elasticsearch

**Comment by Dubrzr** [comment]
Yes, the first one to develop this feature will have the opportunity to choose the import ;) :p
